### PR TITLE
GH2047: Added option to provide version in nuget updation

### PR DESCRIPTION
**Pull request by nikhilagrawaldotnet**
This PR provides option to specify package version while updating and is same as doing

    nuget.exe Update "FooBar.sln" -Version 1.0.0.0

### CODE REVIEW

**Comment by jnm2** [comment]
Wow, a lot of notifications are getting sent! Maybe you could temporarily change the title to the standard `[wip] Added option to...` or `[Do not merge] Added option to...` rather than closing and starting a new PR?

**Comment by jnm2** [comment]
Wow, a lot of notifications are getting sent! Maybe you could temporarily change the title to the standard `[wip] Added option to...` or `[Do not merge] Added option to...` rather than closing and starting a new PR? Even closing and reopening the same PR is better than opening many PRs.

When the PR is open, it automatically updates to match your branch. Force pushing the PR branch to correct the mistakenly included commits is typical. (Hope this is helpful!)

**Comment by gep13** [comment]
@nikhilagrawaldotnet I know that this was spoken about, but was there an issue also created to go along with this PR as per our Contributing Guidelines?  If so, can you link to the issue?

**Comment by nikhilagrawaldotnet** [comment]
@gep13 Thanks for the headsup. I have left a message for @devlead on Gitter. Further to his comment will follow the issue -> PR route. Till then renamed PR to `[Do not merge]...`.

File: src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs [hunk 1]
```diff
@@ -53,5 +54,11 @@ public sealed class NuGetUpdateSettings : ToolSettings
         /// </summary>
         /// <value>The version of MSBuild to be used with this command.</value>
         public NuGetMSBuildVersion? MSBuildVersion { get; set; }
+
+        /// <summary>
+        /// Gets or sets package version to be used with this command.
+        /// </summary>
+        /// <value>The package version to be used with this command.</value>
+        public Version Version { get; set; }
```
Comment (mholo65) [inline src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs]: Don't think `System.Version` is the best type here as it don't supports SemVer, which is required if one would like to use a prerelease version.

Comment (mholo65) [reply src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs]: Along with this, we should also probably add `PreRelease` argument as well as it goes hand in hand with the `Version` argument.

Comment (nikhilagrawaldotnet) [reply src/Cake.Common/Tools/NuGet/Update/NuGetUpdateSettings.cs]: ah. Thanks @mholo65 for the insight. Changed `Version` prop to string to accommodate SemVer. The `PreRelease` option is already there.

**Comment by nikhilagrawaldotnet** [comment]
@mholo65 @devlead Can you please merge this so I can pick alpha from myget. Tx

**Comment by gep13** [comment]
@nikhilagrawaldotnet the focus of the next release is getting a working, stable, version that targets .Net Core 2.0.  As a result, this PR likely won't be merged until we start looking at 0.27.0.  The best suggestion would be to use the artifacts that are generated as part of the AppVeyor build for your PR, which you can find here:

https://ci.appveyor.com/project/cakebuild/cake/build/0.26.0-PullRequest.2046+7.build.4310/artifacts

**Comment by nikhilagrawaldotnet** [comment]
Can we close this pls now that 26 is live.

**Comment by gep13** [comment]
@nikhilagrawaldotnet now that 0.26.0 has shipped, we will be looking to start closing off some of the open PR's, yes.  To help us with that though, if you could rebase your branch against the latest develop branch (instructions are in the contributing guide) that woudl really help.

**Comment by mholo65** [comment]
@nikhilagrawaldotnet I've rebased this and will merge as soon as CI passes.

**Comment by mholo65** [comment]
@nikhilagrawaldotnet Your contributions have been merged. Thank you for your contribution.

**Pull request merged**